app/src/main/java/com/devindi/wallpaper/settings/model/SettingsManager.kt

```diff
@@ -0,0 +1,34 @@
+package com.devindi.wallpaper.settings.model
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.content.SharedPreferences
+import com.devindi.wallpaper.R
+
+const val DIMENSION_HEIGHT = "wallpaper_height"
+const val DIMENSION_WIDTH = "wallpaper_width"
+
+class SettingsManager(storage: SharedPreferences) {
+
+    private val map = HashMap<String, SettingsField<*>>()
+    private val fieldLive = MutableLiveData<String>()
+    private val sharedPreferenceChangeListener = SharedPreferences.OnSharedPreferenceChangeListener {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/devindi/mapwall/blob/264baa37465d6b2f1e3fbd000c3445362f879c9d/app/src/main/java/com/devindi/wallpaper/settings/model/SettingsManager.kt#L15"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>